addon-test-support/@ember/test-helpers/helper-hooks.ts

```diff
@@ -1,6 +1,23 @@
-type Hook = (...args: any[]) => void | Promise<void>;
-type HookLabel = 'start' | 'end' | string;
-type HookUnregister = {
+export type Hook = (...args: any[]) => void | Promise<void>;
+export type HookHelperName =
+  | 'blur'
+  | 'click'
+  | 'doubleClick'
+  | 'fillIn'
+  | 'fireEvent'
+  | 'focus'
+  | 'render'
+  | 'scrollTo'
+  | 'select'
+  | 'tab'
+  | 'tap'
+  | 'triggerEvent'
+  | 'triggerKeyEvent'
+  | 'typeIn'
+  | 'visit'
+  | string;
+export type HookLabel = 'start' | 'end' | string;
```

Comment:
I recognize this one in particular was here previously, and so misled you, but it turns out that this type definition is basically meaningless: it's just `string`. Here's why: types in TypeScript are best thought of as *sets*, where union types (`A | B`) are the *union of the sets*. The set represented by `| string` at the end of this type *includes* the strings `'start'` and `'end'` (and all other strings):

![image](https://user-images.githubusercontent.com/2403023/222757028-4437b6c1-50d3-4ce7-9bba-ae99d53d6c35.png)


This means that saying `'start' | 'end' | string` is *exactly* identical from a type-checking perspective to `string`.

In this case, that means we can remove this type definition entirely, along with its corresponding export. It also means we'll want to do the same on the newly-introduced `HookHelperName` definition above, which has the same limitation!

```suggestion
```

Reply:
Are 'start' and 'end' the most common labels that will be used? I assumed this was here more as documentation rather than expecting it to do anything re: type-checking. If we switch to just `string`, we should ensure that the documentation makes the 'start' and 'end' parts obvious.

Reply:
They are the *defaults*, and in fact that's already documented at the point of use of the parameter in the function. It's possible someone intended it as "documentation" of sorts, but it tends very much to trip up folks new to TS and even if it was *intended* that way, we shouldn't *leave* it that way so that people don't get the wrong idea.

Reply:
@chriskrycho - you're comment on this ticket presumes that the original author (me) didn't understand how unions work in TS. This isn't the case. We weren't sure how this feature was going to shape up at the time, hence the reason it was private. Adding the `string` operated as a catch all for us, which allowed us the flexibility to do as we wanted without having to update this library constantly. 

So, it's not useless. Hooks evolved to only be used for `start` and `end`, so if anything, we should just remove the string fallback.

Reply:
Removing the `string` fallback makes good sense if it's now known that's all it should support! 👍🏼

Reply:
In addition to `start` and `end`, we also have `targetFound`. Included it and removed the `string` fallback.